github/github_test.go

```diff
@@ -42,8 +31,8 @@ const (
 // setup sets up a test HTTP server along with a github.Client that is
 // configured to talk to that test server. Tests should register handlers on
 // mux which provide mock responses for the API method being tested.
-func setup() {
-	// test server
+func setup() (serverURL string, mux *http.ServeMux, client *Client, teardown func()) {
```

Comment:
I've looked over the PR, and noticed that the vast majority of instances ignore `serverURL`. In fact, it's only used in 1 place.

`client` and teardown are the most often used parameters, used in all instances.

`mux` is used in most, but missing in about 20-30% of them.

Since `setup` is going to be called in so many places and in all future tests, let's optimize their order based on frequency of use. I think the following order would be better (most often used parameters first, except `teardown` is always last):

```Go
func setup() (client *Client, mux *http.ServeMux, serverURL string, teardown func())
```

I just think that's going to look nicer than the `_, mux, client, teardown := setup()` lines that we get now.

Reply:
Thanks for the feedback, @shurcooL. This is a very valid point, and something that is going to be used so often deserves this kind of depth of designing for sure. 

I will make this change and also merge in any new tests again. Unfortunately I am unable to work on this until next week - but I will endeavour to get it done then.